website/src/components/Navigation/NavigationTab.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+interface NavigationTabProps {
+    isActive?: boolean;
+    children: React.ReactNode;
+    as?: 'button' | 'a';
+    href?: string;
+    onClick?: () => void;
+    className?: string;
+}
```

Comment:
```suggestion
type NavigationTabProps = {
    isActive?: boolean;
    children: React.ReactNode;
    as?: 'a';
    href?: string;
    className?: string;
} |  {
    isActive?: boolean;
    children: React.ReactNode;
    as?: 'button';
    onClick?: () => void;
    className?: string;
}
```

Would that be a cleaner typing? This would allow removing the check `&& href` below which I found a bit strange as it led a `button` to be used if no `href` is provided even if the component user explicitly sets `as='a'`.

Reply:
@claude try to implement this